### PR TITLE
Get rid of `NetworkService` in `ChainSync`

### PR DESCRIPTION
**Pull request by dmitry-markin**
Move peer banning from `ChainSync` to `SyncingEngine`.

### CODE REVIEW

**Comment by dmitry-markin** [comment]
@altonen To give a heads up, there is one final change coming after this PR is merged — unifying all `ChainSync` actions under one enum and deferring the action processing until after `select!`, as you proposed some time ago:
```rust
tokio::select! {
    ...
}

if let Some(action) = self.chain_sync.take_action() {
    self.perform_chain_sync_action(action);
}
```
As the code is organized now, there is maximum one `ChainSync` action pending after `select!`.

File: substrate/client/network/sync/src/engine.rs [hunk 1]
```diff
@@ -1257,14 +1262,30 @@ where
...
+								self.block_announce_protocol_name.clone(),
+							);
+							self.network_service.report_peer(peer_id, rep);
+						},
+						OnStateResponse::Nothing => {},
 					}
 				},
 				PeerRequest::WarpProof => {
-					self.chain_sync.on_warp_sync_response(peer_id, EncodedProof(resp));
+					match self.chain_sync.on_warp_sync_response(&peer_id, EncodedProof(resp)) {
```
Comment (altonen) [inline substrate/client/network/sync/src/engine.rs:1279]: `if let Err(BadPeer(peer_id, rep)) = ...`

File: substrate/client/network/sync/src/chain_sync.rs [hunk 2]
```diff
@@ -1918,7 +1897,13 @@ where
 		}
 	}
 
-	fn on_warp_sync_data(&mut self, who: &PeerId, response: EncodedProof) -> Result<(), BadPeer> {
+	/// Submit a warp proof response received.
+	#[must_use]
+	pub fn on_warp_sync_response(
+		&mut self,
+		who: &PeerId,
```
Comment (altonen) [inline substrate/client/network/sync/src/chain_sync.rs:1904]: `peer_id`

**Review by altonen — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

File: substrate/client/network/sync/src/engine.rs (hunk 1 above)
Comment (dmitry-markin) [reply substrate/client/network/sync/src/engine.rs]: Done

File: substrate/client/network/sync/src/chain_sync.rs (hunk 2 above)
Comment (dmitry-markin) [reply substrate/client/network/sync/src/chain_sync.rs]: Done in the entire file)

**Pull request merged**